OptimizelySDK.Tests/NotificationTests/NotificationCenterTests.cs

```diff
@@ -0,0 +1,195 @@
+            Assert.AreEqual(3, NotificationCenter.NotificationsCount);
+
+            // Verify that ClearAllNotifications remove all the callbacks.
+            NotificationCenter.ClearAllNotifications();
+            Assert.AreEqual(0, NotificationCenter.NotificationsCount);
+        }
+
+        [Test]
+        public void TestFireNotifications()
+        {
+            var decisionNotification = NotificationCenter.NotificationType.Decision;
+            var trackNotification = NotificationCenter.NotificationType.Track;
+            var config = ProjectConfig.Create(TestData.Datafile, LoggerMock.Object, new NoOpErrorHandler());
+
+            // Mocking notification callbacks.
+            var notificationCallbackMock = new Mock<TestNotificationCallbacks>();
+            notificationCallbackMock.Setup(nc => nc.TestDecisionCallback(It.IsAny<Experiment>(), It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), It.IsAny<Variation>(), It.IsAny<LogEvent>()));
+            notificationCallbackMock.Setup(nc => nc.TestAnotherDecisionCallback(It.IsAny<Experiment>(), 
+                It.IsAny<string>(), It.IsAny<UserAttributes>(), It.IsAny<Variation>(), It.IsAny<LogEvent>()));
+
+            // Adding decision notifications.
+            NotificationCenter.AddNotification(decisionNotification, notificationCallbackMock.Object.TestDecisionCallback);
+            NotificationCenter.AddNotification(decisionNotification, notificationCallbackMock.Object.TestAnotherDecisionCallback);
+
+            // Adding track notifications.
+            NotificationCenter.AddNotification(trackNotification, notificationCallbackMock.Object.TestTrackCallback);
+            
+            // Firing decision type notifications.
+            NotificationCenter.FireNotifications(decisionNotification, config.GetExperimentFromKey("test_experiment"), "testUser",
+                new UserAttributes(), config.GetVariationFromId("test_experiment", "7722370027"), null);
+
+            // Verify that only the registered notifications of decision type are called.
+            notificationCallbackMock.Verify(nc => nc.TestDecisionCallback(It.IsAny<Experiment>(), It.IsAny<string>(), 
+                It.IsAny<UserAttributes>(), It.IsAny<Variation>(), It.IsAny<LogEvent>()), Times.Exactly(1));
+            notificationCallbackMock.Verify(nc => nc.TestAnotherDecisionCallback(It.IsAny<Experiment>(), It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), It.IsAny<Variation>(), It.IsAny<LogEvent>()), Times.Exactly(1));
+            notificationCallbackMock.Verify(nc => nc.TestTrackCallback(It.IsAny<string>(), It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), It.IsAny<EventTags>(), It.IsAny<LogEvent>()), Times.Never);
+        }
+    }
+
+    #region Test Notification callbacks class.
+
+    /// <summary>
+    /// Test class containing dummy notification callbacks.
+    /// </summary>
+    public class TestNotificationCallbacks
```

Comment:
I think this way is just OK but I would recommend to define methods using delegate. 
e.g.
var TestDecisionCallback = delegate(Experiment exp,.... )
Don't spend too much time but it's just an idea.

Reply:
Ok. Will discuss it with you in the morning.